landoapi/phabricator_client.py

```diff
@@ -0,0 +1,109 @@
+        self.api_url = os.getenv('PHABRICATOR_URL') + '/api'
+        self.api_key = api_key
+
+    def get_revision(self, id=None, phid=None):
+        """ Gets a revision as defined by the Phabricator API.
+        
+        Args:
+            id: The id of the revision if known. This can be in the form of 
+                an integer or an integer prefixed with 'D', e.g. 'D12345'.
+            phid: The phid of the revision to be used if the id isn't provided. 
+        
+        Returns: 
+            A hash of the revision data just as it is returned by Phabricator.
+            Returns None, if the revision doesn't exist, or if the api key that 
+            was used to create the PhabricatorClient doesn't have permission to 
+            view the revision. 
+        """
+        if id:
+            id_num = str(id).strip().replace('D', '')
+            result = self.__get('/differential.query', {'ids[]': [id_num]})
+        elif phid:
+            result = self.__get('/differential.query', {'phids[]': [phid]})
+        else:
+            return None
+        return result[0] if result else None
+
+    def get_current_user(self):
+        """ Gets the information of the user making this request.
+        
+        Returns:
+            A hash containing the information of the user that owns the api key
+            that was used to initialize this PhabricatorClient.
+        """
+        return self.__get('/user.whoami')
+
+    def get_user(self, phid):
+        """ Gets the information of the user based on their phid. 
+        
+        Args:
+            phid: The phid of the user to lookup.
+        
+        Returns:
+            A hash containing the user information, or an None if the user 
+            could not be found.
+        """
+        result = self.__get('/user.query', {'phids[]': [phid]})
+        return result[0] if result else None
+
+    def get_repo(self, phid):
+        """ Get basic information about a repo based on its phid. 
+        
+        Args:
+            phid: The phid of the repo to lookup.
+            
+        Returns:
+            A hash containing the repo info, or None if the repo isn't found.
+        """
+        result = self.__get('/phid.query', {'phids[]': [phid]})
+        return result.get(phid) if result else None
+
+    def __request(self, url, data=None, params=None, method='GET'):
+        data = data if data else {}
+        data['api.token'] = self.api_key
+        response = requests.request(
+            method=method,
+            url=self.api_url + url,
+            params=params,
+            data=data,
+            timeout=10
+        ).json()
+
+        if response['error_code']:
+            exp = PhabricatorAPIException(response.get('error_info'))
+            exp.error_code = response.get('error_code')
+            exp.error_info = response.get('error_info')
+            raise exp
+
+        return response.get('result')
+
+    def __get(self, url, data=None, params=None):
```

Comment:
This method name should be preceded by a single underscore.  I also find overloading the name `get()` a bit of a mind twist - doing so is rare in Python.  You might want to use a different method name, like `_do_get()` or `_GET()`.